daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DarReader.scala

```diff
@@ -98,35 +79,30 @@ object GenDarReader {
   private val ManifestName = "META-INF/MANIFEST.MF"
   private[archive] val EntrySizeThreshold = 1024 * 1024 * 1024 // 1 GB
 
-  private[archive] case class ZipEntries(name: String, entries: Map[String, (Long, InputStream)]) {
+  private[archive] case class ZipEntry(size: Long, getStream: () => InputStream)
 
-    def getInputStreamFor(entryName: String): Try[(Long, InputStream)] = {
+  private[archive] case class ZipEntries(name: String, entries: Map[String, Bytes]) {
+    private[GenDarReader] def get(entryName: String): Try[Bytes] = {
       entries.get(entryName) match {
-        case Some((size, is)) => Success(size -> is)
+        case Some(is) => Success(is)
         case None => Failure(Error.InvalidZipEntry(entryName, this))
       }
     }
 
-    def readDalfNames(
-        readDalfNamesFromManifest: InputStream => Try[Dar[String]]
-    ): Try[Dar[String]] =
-      parseDalfNamesFromManifest(readDalfNamesFromManifest).recoverWith { case NonFatal(e1) =>
-        findLegacyDalfNames().recoverWith { case NonFatal(_) =>
-          Failure(Error.InvalidDar(this, e1))
+    private[GenDarReader] def readDalfNames: Try[Dar[String]] =
+      bracket(get(ManifestName).map(_.toInputStream))(is => Try(is.close()))
+        .flatMap(DarManifestReader.dalfNames)
+        .recoverWith { case NonFatal(e1) =>
+          findLegacyDalfNames.recoverWith { case NonFatal(_) =>
+            Failure(Error.InvalidDar(this, e1))
+          }
```

Comment:
@cocreature Do we have to handle dar without Manifest ? 
If no, let me drop this code, otherwise should we have test for that (It seems we currently do not have any) ?

Reply:
The compiler will always add a manifest and I’d generally consider anything not produced by the compiler to be unsupported so should be fine to drop it.

Reply:
Great !

Reply:
:fire: